### PR TITLE
Change the import config to use exported deps

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

File: rules/go_rules.build_defs [hunk 1]
```diff
@@ -108,8 +112,9 @@ def go_library(name:str, srcs:list, asm_srcs:list=None, hdrs:list=None, out:str=
             building_description = 'Packing...',
             requires = ['go'],
             labels = labels,
-            provides = provides,
+            provides =  {'go': ':' + name, 'go_src': lib_rule},
             test_only = test_only,
+            exported_deps = exported_deps,
```
Comment (sagikazarmark) [inline rules/go_rules.build_defs:117]:
```suggestion
            exported_deps = exported_deps + deps,
```

Not really nice, but fixes #1181

**Comment by sagikazarmark** [comment]
Not related to this PR directly, but this seems weird:

```
    cmd = f'echo packagefile {import_path}/{name}=$PKG_DIR/{out} > $OUT'
    if pkg.endswith(name):
        cmd = f'{cmd} && echo packagefile {import_path}=$PKG_DIR/{out} >> $OUT'
```

Why do we need to append the package name to the import path? If it's provided, it should just be the right import path IMHO.

**Comment by Tatskaari** [comment]
> Not related to this PR directly, but this seems weird:
> 
> ```
>     cmd = f'echo packagefile {import_path}/{name}=$PKG_DIR/{out} > $OUT'
>     if pkg.endswith(name):
>         cmd = f'{cmd} && echo packagefile {import_path}=$PKG_DIR/{out} >> $OUT'
> ```
> 
> Why do we need to append the package name to the import path? If it's provided, it should just be the right import path IMHO.
> 
> This could actually be an issue if you have a package nested inside another with the same name.

To maintain the existing behavior (we used to symlink all the .a files back one directory) I will try and remove this for v16.

File: rules/go_rules.build_defs (hunk 1 above)
Comment (Tatskaari) [reply rules/go_rules.build_defs:117]: Nice spot! They don't need to be exported, they just need to be deps which is nice.

**Review by peterebden — APPROVED** [review]
I think this makes sense. One thing to check is that it doesn't explode the fan-out too much in terms of dependencies needed to build each action?

**Pull request merged**